stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/entity/AttachmentEntity.kt

```diff
@@ -0,0 +1,53 @@
+package io.getstream.chat.android.livedata.entity
+
+import androidx.room.ColumnInfo
+import androidx.room.Embedded
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.PrimaryKey
+
+@Entity(
+    tableName = "attachment_inner_entity",
+    foreignKeys = [
+        ForeignKey(
+            entity = MessageInnerEntity::class,
+            parentColumns = ["id"],
+            childColumns = ["messageId"],
+            onDelete = ForeignKey.CASCADE,
+            deferred = true
+        )
+    ]
+)
+internal data class AttachmentEntity(
+    @ColumnInfo(index = true)
+    val messageId: String,
+    val authorName: String?,
+    val titleLink: String?,
+    val thumbUrl: String?,
+    val imageUrl: String?,
+    val assetUrl: String?,
+    val ogUrl: String?,
+    val mimeType: String?,
+    val fileSize: Int,
+    val title: String?,
+    val text: String?,
+    val type: String?,
+    val image: String?,
+    val url: String?,
+    val name: String?,
+    val fallback: String?,
+    val uploadFilePath: String?,
+    @Embedded val uploadState: UploadStateEntity?,
+    val extraData: Map<String, Any>,
+) {
+    @PrimaryKey
+    var id: Int = hashCode()
```

Comment:
@PrimaryKey(autoGenerate = true) ?

Reply:
Not sure. I consider Entity as a data class. So you can compare two instances by equal autogenerated code of kotlin data classes, so hashCode of a data class is a unique id, which is equal for two equal instances.